modules/nux/modules.php

```diff
@@ -260,47 +273,264 @@ public function process() {
     }
 }
 
+/**
+ * @subpackage nux/handler
+ */
+class Hm_Handler_process_import_accouts_servers extends Hm_Handler_Module
+{
+    public function process()
+    {
+        list($success, $form) = $this->process_form(array('accounts_source'));
+
+        if ($success) {
+            $file = $this->request->files['accounts_sample'];
+            if (!is_null($file) && $file['type'] == 'application/x-yaml') {
+                try {
+                    $servers = Yaml::parseFile($file['tmp_name']);
+                } catch (\Throwable $th) {
+                    Hm_Msgs::add('ERR' . $th->getMessage());
+                    $this->save_hm_msgs();
+                    Hm_Dispatch::page_redirect('?page=servers');
+                }
+                foreach ($servers as $key => $value) {
+                    if (empty($value['user']['username']) or empty($value['user']['password'])) {
+                        Hm_Msgs::add('ERRUsername and password are required for: ' . $key . ' server');
+                        return;
+                    }
+                    if ($value['jmap'] && !empty($value['jmap']['jmap_server'])) {
+                        if (!$this->module_is_supported('jmap')) {
+                            Hm_Msgs::add("ERRJMAP module is not enabled");
+                            return;
```

Comment:
For bulk import we shouldn't return. Also we must check for supported modules once not in the loop to avoid sending multiple times one message.